proto/queue/queue.proto

```diff
@@ -23,6 +23,30 @@ message ControlEvent {
     ExecuteResponse execute_response = 3;
     DeliverablePrefixes deliverable_prefixes = 4;
     UndeliverablePrefixes undeliverable_prefixes = 5;
+    CloseJob close_job = 6;
+    Close close = 7;
+  }
+}
+
+// JobEvent is are queue events which should be processed by the inner Job
+// event loop. These events are performed on per the per Job handler.
```

Comment:
nit: These events are performed on the per-job handler

not sure if my suggestion the best one grammatically